contracts/common/lib/MaxEffectiveBalanceIncreaser.sol

```diff
@@ -0,0 +1,138 @@
+// SPDX-FileCopyrightText: 2025 Lido <info@lido.fi>
+// SPDX-License-Identifier: GPL-3.0
+
+/* See contracts/COMPILERS.md */
+// solhint-disable-next-line lido/fixed-compiler-version
+pragma solidity >=0.8.9 <0.9.0;
+
+/**
+ * @title A lib for EIP-7251: Increase the MAX_EFFECTIVE_BALANCE.
+ * Allow validators to have larger effective balances, while maintaining the 32 ETH lower bound.
+ */
+contract MaxEffectiveBalanceIncreaser {
+    address public constant CONSOLIDATION_REQUEST_PREDEPLOY_ADDRESS = 0x0000BBdDc7CE488642fb579F8B00f3a590007251;
+
+    uint256 internal constant PUBLIC_KEY_LENGTH = 48;
+    uint256 internal constant CONSOLIDATION_REQUEST_CALLDATA_LENGTH = PUBLIC_KEY_LENGTH * 2;
+
+    error ConsolidationFeeReadFailed();
+    error ConsolidationFeeInvalidData();
+    error ConsolidationFeeRefundFailed(address recipient, uint256 amount);
+    error ConsolidationRequestAdditionFailed(bytes callData);
+    error NoConsolidationRequests();
+    error MalformedPubkeysArray();
+    error MalformedTargetPubkey();
+    error MismatchingSourceAndTargetPubkeysCount(uint256 sourcePubkeysCount, uint256 targetPubkeysCount);
+    error InsufficientValidatorConsolidationFee(uint256 provided, uint256 required);
+    error ZeroArgument(string argName);
+
+    event ConsolidationRequestsTriggered(address indexed sender, bytes[] sourcePubkeys, bytes[] targetPubkeys, address indexed refundRecipient, uint256 excess);
+
+    /**
+     * @dev Send EIP-7251 consolidation requests for the specified public keys.
+     *      Each request instructs a validator to consolidate its stake to the target validator.
+     *
+     * @param _sourcePubkeys An array of tightly packed arrays of 48-byte public keys corresponding to validators requesting consolidation.
+     *      | ----- public key (48 bytes) ----- || ----- public key (48 bytes) ----- | ...
+     *      | ----- public key (48 bytes) ----- || ----- public key (48 bytes) ----- | ...
+     *      | ----- public key (48 bytes) ----- || ----- public key (48 bytes) ----- | ...
+     *
+     * @param _targetPubkeys An array of 48-byte public keys corresponding to validators to consolidate to.
+     *      | ----- public key (48 bytes) ----- || ----- public key (48 bytes) ----- | ...
+     *
+     * @param _refundRecipient The address to refund the excess consolidation fee to.
+     *
+     */
+    function addConsolidationRequests(bytes[] calldata _sourcePubkeys, bytes[] calldata _targetPubkeys, address _refundRecipient) external payable {
```

Comment:
maybe `make` or `trigger` instead of `add` in `addConsolidationRequests`

Reply:
add is good imo, since it has a queue on the EL side, and, more important, EIP-7251 has the same naming:

```solidity
function addConsolidation(bytes memory srcPubkey, bytes memory targetPubkey, uint64 requestFeeLimit) private {
```